File: sportskirezultati/src/main/java/com/sportskirezultati/auth/AuthService.java
package com.sportskirezultati.auth;


import com.sportskirezultati.auth.dto.LoginRequest;
import com.sportskirezultati.auth.dto.LoginResponse;
import com.sportskirezultati.auth.dto.LogoutResponse;
import com.sportskirezultati.auth.dto.RefreshRequest;
import com.sportskirezultati.auth.dto.RefreshResponse;
import com.sportskirezultati.auth.dto.RegisterRequest;
import com.sportskirezultati.auth.dto.RegisterResponse;

/** Service for authentication. */
public interface AuthService {

  /** Check credentials and login */
  LoginResponse login(LoginRequest request);

  /** Check if unique username and password and save user. */
  RegisterResponse signUp(RegisterRequest request);

  /** Return new jwt token. */
  RefreshResponse refreshToken(RefreshRequest request);

  /** Logout user. */
  LogoutResponse logout(Long userId);
}

File: sportski-rezultati-angular/src/app/auth/auth.service.ts
import { HttpHeaders, HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { RegisterRequestDto } from '../aauth/dto/register-request-dto';
import { LoginResponseDto } from './dto/login-response-dto';
import { StorageService } from './storage.service';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  constructor(private httpClient: HttpClient, private storageService: StorageService) { }


  login(username: string | any, password: string | any): Observable<any> {
    return this.httpClient.post(environment.baseUrl + '/api/auth/signin', {
      username,
      password
    }, httpOptions);
  }

  register(registerRequest: RegisterRequestDto): Observable<any> {
    return this.httpClient.post(environment.baseUrl + '/api/auth/signup', registerRequest, httpOptions);
  }

  refreshToken(token: string) {
    return this.httpClient.post(environment.baseUrl + '/api/auth/refreshtoken', {
      refreshToken: token
    }, httpOptions);
  }
}

